### PR TITLE
Fix nested subscription calls

### LINKED ISSUE #20442: Double number of extension views in Code Insights page.
#### Steps to reproduce:

1. Open Sourcegraph.com and go to the settings page.
2. Change any settings in settings file.
3. Go back to `/insights` page.
4. Open dev tool and see that we have double numbers of code insight views. (See screenshot below)

image

Original handler where we fetch/get code insights views - https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/client/web/src/insights/pages/InsightsPage.tsx#L22-27

#### Expected behavior:
After changing user or organisation config/settings we **don't have** double number of code insights.

#### Actual behavior:
After changing user or organisation config/settings we **have** double number of code insights.

### PR DESCRIPTION
**Pull request by vovakulikov**
Closes: https://github.com/sourcegraph/sourcegraph/issues/20442

This PR is about removing explicit nested calls of context subscription. Nested subscriptions lead to problems in a consumer such as a double amount of view providers because we do not unsubscribe from the subscriptions of previous setting update-event. [More context in this comment](https://github.com/sourcegraph/sourcegraph-code-stats-insights/pull/5#discussion_r627721610)

### CODE REVIEW

File: src/search-insights.ts [hunk 1]
```diff
@@ -49,8 +49,21 @@ export function activate(context: sourcegraph.ExtensionContext): void {
...
+    // subscription bag needed to unsubscribe nested calls in a moment when user/org settings
+    // have been updated.
+    let previousSubscriptions = new Subscription()
+
     context.subscriptions.add(
-        insightChanges.pipe(mergeAll()).subscribe(([id, insight]) => {
+        insightChanges.pipe(
+            tap(() => {
+                previousSubscriptions.unsubscribe()
+                previousSubscriptions = new Subscription()
```
Comment (felixfbecker) [inline src/search-insights.ts:62]: I wonder if this line (instantiating a new instance) is even necessary 🤔

**Review by felixfbecker — APPROVED** [review]
Nice!

File: src/search-insights.ts (hunk 1 above)
Comment (vovakulikov) [reply src/search-insights.ts:62]: Yes, because if you unsubscribed once then a next subscriptions will be immediately unsubscribed as well

Comment (felixfbecker) [reply src/search-insights.ts:62]: Oh right

**Pull request merged**